rfd/0122-kube-offline-access.md

```diff
@@ -0,0 +1,163 @@
+append the participants to the session. On the other hand, `kubectl exec` (non-interactive sessions) does not require a permanent
+connection to Auth because the session is created and closed immediately after the command completes and the session tracker
+is never created.
+
+Without Auth connectivity, it is not possible to create or join a moderated session.
+
+#### Web UI
+
+Not applicable to this RFD as the Web UI is not supported for Kubernetes clusters.
+
+#### Strict Locking mode
+
+```yaml
+kind: role
+version: v5
+metadata:
+  name: lock-strict
+spec:
+  options:
+    lock: strict
+```
+
+When Strict locking mode is enabled, Teleport requires a connection to Auth to ensure that the locks are up to date.
+If the locks are not up to date, Teleport will terminate all connections to the cluster and prevent new connections from being established.
+
+Strict locking isn't enabled by default.
+
+### Auth Connection Not Required
+
+### `kubectl`
+
+Attempting to connect to a Kubernetes cluster when the user's certificate is cached locally and does not require MFA should not require a connection to Auth.
+
+The exception to this is when the user tries to create an interactive session (`kubectl exec -it`) because the session is moderated and requires a connection to Auth.
+
+###### Per-Session MFA
+
+With Teleport 13, `tsh proxy kube` is now able to cache MFAs certificates for longer than 1 minute. This means that if a user has already
+completed the MFA ceremony within the `max_session_ttl` time, `tsh proxy kube` will not need to connect to Auth to perform the MFA ceremony again
+until the `max_session_ttl` has expired.
+
+This feature is useful when Auth server downtime is expected and users can complete the MFA ceremony before Auth downtime.
+
+###### Session Trackers
+
+A new session tracker is created for every interactive session. Kubernetes Agent will attempt to persist the session tracker
+but if it cannot, the session will be aborted.
+
+We propose the same solution as RFD 93 to mitigate the impact of Auth downtime on interactive sessions.
+If the session tracker cannot be persisted, the agent should check if moderated sessions are required.
+If they aren't, the session should be allowed to proceed using a local session tracker resource.
+
+While this does allow access to the node, it will prevent the following:
+
+- the session from appearing in the active sessions list
+- the session will not be joinable by other users
+- the session recording will not be available until the UploadCompleter processes it
+
+### Teleport Connect
+
+Currently, Teleport Connect leverages the `tsh` client and kubectl to establish a connection to the cluster which means that
+the MFA certificates are cached locally but only for 1 minute.
+
+We have plans to move Teleport Connect to `tsh proxy kube` which will allow us to leverage the longer MFA certificate caching
+and allow users to complete the MFA ceremony before Auth downtime.
```

Comment:
This is not yet implemented and it's just a future consideration,

When I spoke with @ravicious about it, he mentioned that it would take some effort.